source/MRMesh/MRMeshLoadObj.cpp

```diff
@@ -143,141 +146,158 @@ tl::expected<std::vector<NamedMesh>, std::string> fromSceneObjFile( const char*
 
     Timer timer( "split by lines" );
     std::vector<size_t> newlines{ 0 };
-    tbb::enumerable_thread_specific<std::vector<size_t>> newlinesPerThread;
-    tbb::parallel_for( tbb::blocked_range<size_t>( 0, size ), [&] ( const tbb::blocked_range<size_t>& range )
     {
-        bool exists = false;
-        auto& newlines = newlinesPerThread.local( exists );
-        // blocks shall not intersect each other
-        assert( !exists );
-        for ( auto ci = range.begin(); ci < range.end(); ci++ )
+        constexpr size_t blockSize = 4096;
+        const auto blockCount = (size_t)std::ceil( (float)size / blockSize );
+        constexpr size_t maxGroupCount = 48;
```

Comment:
may be
```
maxGroupCount = tbb::global_control::active_value( tbb::global_control::max_allowed_parallelism )

Reply:
This value is not optimal, as seen by benchmarks.